core/runtime.rs

```diff
@@ -270,6 +272,11 @@ pub struct RuntimeOptions {
   /// The snapshot is deterministic and uses predictable random numbers.
   pub will_snapshot: bool,
 
+  /// An optional callback that will be called for each module that is loaded
+  /// during snapshotting. This callback can be used to transpile source on the
+  /// fly, during snapshotting, eg. to transpile TypeScript to JavaScript.
+  pub snapshot_module_load_cb: Option<InternalModuleLoaderCb>,
```

Comment:
Is this ever used for anything but transpiling? Why not make this a boolean?

Reply:
Because `deno_core` is agnostic of TypeScript and doesn't depend on any crates that support TypeScript. This is done in a way that embedder can inject whatever callback they want. Do you want me to add a dependency on `deno_ast` to `deno_core`?

Reply:
makes sense